### PR TITLE
Update Registration Contract in Solidity

### LINKED ISSUE #911: Update VRC chain initialization and receipt root tracking
We need to update VRC to include the latest chain initialization logic and receipt root tree for tracking individual deposit log. This is a must-have for our testnet. I will work on this during/after Thanksgiving

### PR DESCRIPTION
**Pull request by terenc3t**
This resolves #911

We update our validator registration contract in Solidity for ETH 1.0 Chain. The contract spec was written [here](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#pow-chain-registration-contract)

Note, we are currently using inline assembly for byte operations. These operations should be optimized and thoroughly tested if we were ever going to use this contract in production to handle real eth. Without going down to the rabbit hole on solidity inline assembly, I opened #1008 for whoever is interested to pick it up.

- [x] Update contract
- [x] Update readme markdown
- [x] Update test

### CODE REVIEW

File: .soliumrc.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
       "error",
       "double"
     ],
+    "security/no-inline-assembly": ["warning"],
```
Comment (prestonvanloon) [inline .soliumrc.json]: Can you explain this change in the PR description please?

File: contracts/validator-registration-contract/BUILD.bazel [hunk 2]
```diff
@@ -6,12 +6,12 @@ go_library(
     importpath = "github.com/prysmaticlabs/prysm/contracts/validator-registration-contract",
     visibility = ["//visibility:public"],
     deps = [
-        "//shared/event:go_default_library",
         "@com_github_ethereum_go_ethereum//:go_default_library",
         "@com_github_ethereum_go_ethereum//accounts/abi:go_default_library",
         "@com_github_ethereum_go_ethereum//accounts/abi/bind:go_default_library",
         "@com_github_ethereum_go_ethereum//common:go_default_library",
         "@com_github_ethereum_go_ethereum//core/types:go_default_library",
+        "@com_github_ethereum_go_ethereum//event:go_default_library",
```
Comment (prestonvanloon) [inline contracts/validator-registration-contract/BUILD.bazel]: Please revert back to the shared library.

File: .soliumrc.json (hunk 1 above)
Comment (terenc3t) [reply .soliumrc.json]: Will update the description and create an issue to track

File: contracts/validator-registration-contract/BUILD.bazel (hunk 2 above)
Comment (terenc3t) [reply contracts/validator-registration-contract/BUILD.bazel]: Done

**Pull request merged**